### PR TITLE
[ML] rename trained model allocations to assignments

### PR DESCRIPTION
**Pull request by benwtrent**
This renames the internal concept of a trained model allocation into an assignment.

Now models are assigned to a node and routes created for inference. Not "allocated". 

This is an internal rename only. The user facing concepts of trained models and deployments are untouched.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/DeleteTrainedModelAssignmentAction.java [hunk 1]
```diff
@@ -18,11 +18,11 @@
 import java.io.IOException;
 import java.util.Objects;
 
-public class DeleteTrainedModelAllocationAction extends ActionType<AcknowledgedResponse> {
-    public static final DeleteTrainedModelAllocationAction INSTANCE = new DeleteTrainedModelAllocationAction();
+public class DeleteTrainedModelAssignmentAction extends ActionType<AcknowledgedResponse> {
+    public static final DeleteTrainedModelAssignmentAction INSTANCE = new DeleteTrainedModelAssignmentAction();
     public static final String NAME = "cluster:internal/xpack/ml/model_allocation/delete";
```
Comment (benwtrent) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/DeleteTrainedModelAssignmentAction.java:23]: Renaming the actions broke WAY too much stuff. So, I didn't rename these internal actions.

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
run elasticsearch-ci/full-bwc

**Comment by benwtrent** [comment]
run elasticsearch-ci/full-bwc

**Comment by benwtrent** [comment]
run elasticsearch-ci/full-bwc

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**